lib/rules/call-indentation.js

```diff
@@ -17,14 +18,91 @@ module.exports = function(context) {
    //--------------------------------------------------------------------------
 
    function validateCallIndentation(node) {
-      var lineIndentationMatches = helper.lineIndentationMatches(node.callee.loc.end.line, node.loc.end.line);
+      var lineIndentationMatches = helper.lineIndentationMatches(node.callee.loc.end.line, node.loc.end.line),
+          endParensLocation = node.loc.end,
+          lastArg = node.arguments[node.arguments.length - 1],
+          firstArg = node.arguments[0],
+          indentAmount = helper.indent,
+          calleeIndent = helper.lineIndent(node.callee.loc.start.line),
+          allArgsNewLine = node.loc.start.line !== firstArg.loc.start.line,
+          prevArg, nodesByStartLine;
 
+      nodesByStartLine = _.reduce(node.arguments, function(memo, arg) {
+         if (memo[arg.loc.start.line]) {
+            memo[arg.loc.start.line].push(arg);
+         } else {
+            memo[arg.loc.start.line] = [ arg ];
+         }
+         return memo;
+      }, {});
+
+      // check if closing parenthesis is correctly indented
       if (!lineIndentationMatches) {
          context.report({
             node: node,
             message: 'Call expressions must begin and end with the same indentation',
          });
       }
+
+      // check if first arg is on a new line
+      if (allArgsNewLine) {
+         _.each(node.arguments, function(arg) {
+            // check that all arguments start on their own line
+            if (nodesByStartLine[arg.loc.start.line].length > 1) {
+               context.report({
+                  node: node,
+                  loc: arg.loc.start,
+                  message: 'Argument should be on a new line',
+               });
+               return;
+            }
+
+            //check indentation
+            if (helper.lineIndent(arg.loc.start.line) !== (calleeIndent + indentAmount)) {
+               context.report({
+                  node: node,
+                  loc: arg.loc.start,
+                  message: 'Expected indent of {{expected}}, found indent of {{found}}',
+                  data: {
+                     expected: calleeIndent + indentAmount,
+                     found: helper.lineIndent(arg.loc.start.line),
+                  },
+               });
+            }
+         });
+
+         // check if closing parens is on a new line
+         if (endParensLocation.line === lastArg.loc.end.line) {
+            context.report({
+               node: node,
+               loc: node.loc.end,
+               message: 'Closing parenthesis should be on a new line',
+            });
+         }
+      } else {
+         _.each(node.arguments, function(arg) {
+            // check that all arguments start on the same line as the previous one
+            // ends on
+            if (prevArg && prevArg.loc.end.line !== arg.loc.start.line) {
```

Comment:
The way I am differentiating between "all on same line" or "all on new line" above is to check whether the first argument is on the same line as the function call start or not.  So, this case would fail saying that b, c, d and the closing parenthesis need to be on a new line.  

I added a test for this case.